### PR TITLE
Create plugin: Update data source templates

### LINKED ISSUE #782: Use grafana/ui components in templates rather than referring to gf class names
See https://github.com/grafana/grafana-plugin-examples/issues/173

Related https://github.com/grafana/grafana-plugin-examples/issues/126

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

Was initially going to replace the usage of `gf-` classnames in forms, but found some more low hanging fruits in the data source templates that I'm fixing. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/plugin-tools/issues/782

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@4.4.0-canary.836.aa6cb3b.0
  npm install @grafana/plugin-e2e@0.24.0-canary.836.aa6cb3b.0
  # or 
  yarn add @grafana/create-plugin@4.4.0-canary.836.aa6cb3b.0
  yarn add @grafana/plugin-e2e@0.24.0-canary.836.aa6cb3b.0
  ```
</details>

### CODE REVIEW

File: docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ export function QueryEditor({ query, onChange, onRunQuery }: Props) {
   const { upperLimit, lowerLimit, tickInterval } = query;
 
   return (
-    <div className="gf-form">
+    <>
```
Comment (sunker) [inline docusaurus/docs/tutorials/build-a-streaming-data-source-plugin.md:82]: Could potentially use the new `Box` component here, but it does not change appearance so don't think we need it.

File: packages/create-plugin/templates/backend/src/datasource.ts [hunk 2]
```diff
@@ -9,6 +9,18 @@ export class DataSource extends DataSourceWithBackend<MyQuery, MyDataSourceOptio
...
 
   getDefaultQuery(_: CoreApp): Partial<MyQuery> {
-    return DEFAULT_QUERY
+    return DEFAULT_QUERY;
+  }
+
+  applyTemplateVariables(query: MyQuery, scopedVars: ScopedVars): Record<string, any> {
+    return {
+      ...query,
+      queryText: getTemplateSrv().replace(query.queryText, scopedVars),
```
Comment (sunker) [inline packages/create-plugin/templates/backend/src/datasource.ts:18]: Regarding scoped variables...currently the ds template for `plugin.json` doesn't specify any [query options](https://grafana.com/developers/plugin-tools/reference-plugin-json#queryoptions). Should we skip passing the scopredVars variable to `getTemplateSrv().replace`?

File: packages/create-plugin/templates/datasource/src/components/ConfigEditor.tsx [hunk 3]
```diff
@@ -39,29 +43,29 @@ export function ConfigEditor(props: Props) {
...
   };
 
-  const { jsonData, secureJsonFields } = options;
-  const secureJsonData = (options.secureJsonData || {}) as MySecureJsonData;
-
   return (
-    <div className="gf-form-group">
-      <InlineField label="Path" labelWidth={12}>
+    <>
+      <InlineField label="Path" labelWidth={14} interactive tooltip={'Json field returned to frontend'}>
```
Comment (sunker) [inline packages/create-plugin/templates/datasource/src/components/ConfigEditor.tsx:48]: Re label width: Opened a [PR](https://github.com/grafana/grafana/pull/84970) in Grafana to increase the width of the name field to 14. That way we at least give the plugin authors a chance to align fields width across the entire page.

**Review by mckn — APPROVED** [review]
LGTM! I think there are some more changes we can do to the config editor but lets do that in a follow up PR. Great work on this!

**Pull request merged**